src/ncp/ncp_uart.cpp

```diff
@@ -297,6 +314,74 @@ void NcpUart::HandleError(otError aError, uint8_t *aBuf, uint16_t aBufLength)
     otNcpStreamWrite(0, reinterpret_cast<uint8_t *>(hexbuf + 1), static_cast<int>(strlen(hexbuf) - 1));
 }
 
+#if OPENTHREAD_ENABLE_NCP_SPINEL_TRANSFORMER
+
+NcpUart::NcpFrameBufferTransformerReader::NcpFrameBufferTransformerReader(NcpFrameBuffer &aTxFrameBuffer) :
+    mTxFrameBuffer(aTxFrameBuffer),
+    mInputBufferLength(0),
+    mOutputBufferReadIndex(0),
+    mOutputBufferLength(0) {}
+
+bool NcpUart::NcpFrameBufferTransformerReader::IsEmpty() const
+{
+    return mTxFrameBuffer.IsEmpty() && !mOutputBufferLength;
+}
+
+otError NcpUart::NcpFrameBufferTransformerReader::OutFrameBegin()
+{
+    otError status = OT_ERROR_FAILED;
+
+    Reset();
+
+    if ((status = mTxFrameBuffer.OutFrameBegin()) == OT_ERROR_NONE)
+    {
+        while (!mTxFrameBuffer.OutFrameHasEnded())
+        {
+            uint8_t byte = mTxFrameBuffer.OutFrameReadByte();
+            mInputBuffer[mInputBufferLength++] = byte;
```

Comment:
`mInputBufferLength` seems to be only used here within this method to track the length of the original frame (before transforming). How about if we make it a local variable instead of a class member variable?

Reply:
Done.